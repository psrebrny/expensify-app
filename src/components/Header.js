import React from 'react';
import { NavLink } from 'react-router-dom';

export const Header = () => {
  return (
      <div>
        <h1>Expensify</h1>
        <NavLink exact to="/" activeClassName="is-active">Home</NavLink><br/>
        <NavLink to="/create" activeClassName="is-active">Create</NavLink><br/>
        <NavLink to="/help" activeClassName="is-active">Help</NavLink>
      </div>
  );
};