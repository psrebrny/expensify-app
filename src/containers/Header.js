import React from 'react';
import { NavLink } from 'react-router-dom';
import { logout } from '../actions/auth';
import { bindActionCreators } from 'redux';
import { connect } from 'react-redux';

export const HeaderClass = (props) => {
  return (
      <div>
        <h1>Expensify</h1>
        <NavLink to="/dashboard" activeClassName="is-active">Dashboard</NavLink>
        <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
        <button onClick={() => {
          props.logout();
        }}>Logout
        </button>
      </div>
  );
};

const mapDispatchToProps = (dispatch) => {
  return bindActionCreators({
    logout
  }, dispatch);
};

export const Header = connect(null, mapDispatchToProps)(HeaderClass);