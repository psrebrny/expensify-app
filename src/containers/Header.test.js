import React from 'react';
import { shallow } from 'enzyme';
import { HeaderClass } from './Header';

test('render header', () => {
  const wrapper = shallow(<HeaderClass logout={() => {
  }}/>);
  expect(wrapper).toMatchSnapshot();
});

test('call logout on click', () => {
  const logout = jest.fn();
  const wrapper = shallow(<HeaderClass logout={logout}/>);
  wrapper.find('button').simulate('click');
  expect(logout).toHaveBeenCalled();
});