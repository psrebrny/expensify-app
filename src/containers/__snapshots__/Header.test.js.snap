// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render header 1`] = `
<div>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="is-active"
    ariaCurrent="true"
    to="/create"
  >
    Create Expense
  </NavLink>
  <button
    onClick={[Function]}
  >
    Logout
  </button>
</div>
`;
